libavfilter/dnn/dnn_backend_openvino.c

```diff
@@ -114,6 +118,49 @@ static int get_datatype_size(DNNDataType dt)
     }
 }
 
+static int fill_vaapi_blob(OVModel *ov_model, TaskItem *task, OVRequestItem *request)
+{
+    // if we use vaapi as decoder hardware accelerator. the decoder output frame will be NV12 format.
+    IEStatusCode status;
+    ie_blob_t *nv12_blob = NULL;
+    ie_blob_buffer_t blob_buffer;
+    unsigned int surface_id = (unsigned int)(uintptr_t)(task->in_frame->data[3]);
+    status = ie_blob_make_memory_nv12_from_va_surface(task->in_frame->height, task->in_frame->width, ov_model->va_context, surface_id, &nv12_blob);
+    if (status != OK) {
+        fprintf(stderr, "ERROR ie_blob_make_memory_from_surface status %d, line %d\n", status, __LINE__);
+    }
+    // error will be raised if we using ie_infer_request_get_blob.
+    // so, only ie_infer_request_set_blob can be used to set input data.
+    status = ie_infer_request_set_blob(request->infer_request, task->input_name, nv12_blob);
+    if (status != OK) {
+        fprintf(stderr, "ERROR ie_infer_request_set_blob status %d, line %d\n", status, __LINE__);
+    }
+    
+    if (ov_model->model->func_type == DFT_PROCESS_FRAME && task->out_frame->format == AV_PIX_FMT_VAAPI) {
+        // only dnn_processing filter need to set output tensor, but now we can't enable full gpu pipeline in dnn_processing 
+        // because openvino doesn't support surface memory in output tensor.
+        // ref: https://github.com/openvinotoolkit/openvino/issues/12548
+        ie_blob_t *out_vablob = NULL;
+        unsigned int out_surface_id = (unsigned int)(uintptr_t)(task->out_frame->data[3]);
+
+        status = ie_blob_make_memory_nv12_from_va_surface(task->out_frame->height, task->out_frame->width, ov_model->va_context, out_surface_id, &out_vablob);
+        if (status != OK) {
+            fprintf(stderr, "ERROR ie_blob_make_memory_from_surface status %d, line %d\n", status, __LINE__);
+        }
+        if (out_surface_id != 0) {
+            // openvino doesn't support this because surface memory will be automatically converted to cl_mem.
+            status = ie_infer_request_set_blob(request->infer_request, task->output_names[0], out_vablob); 
+            if (status != OK) {
+                fprintf(stderr, "ERROR ie_infer_request_set_blob status %d, line %d\n", status, __LINE__);
+            }
+        } else {
+            av_log(&ov_model->ctx, AV_LOG_ERROR, "surface id is 0");
```

Comment:
What does the "surface_id is 0" means?

Reply:
Surface id is 0 when this frame is software frame or not initialized correctly. How about this modification?
```suggestion
            av_log(&ov_model->ctx, AV_LOG_ERROR, "Surface id is 0, output frame is not initialized correctly.");
```